esignet-ida115-integration-impl/src/test/java/io/mosip/ida115/esignet/integration/service/IdaAuthenticatorImplTest.java

```diff
@@ -241,44 +330,308 @@ public void doKycExchange_withValidDetails_thenPass() throws Exception {
+					"addressLine1_eng", List.of(Map.of("value", "address-line-1")), "city_eng",
+					List.of(Map.of("value", "my city")), "pinCode", "221024",
+					BioMatchType.FACE.getIdMapping().getIdname(), faceBdb);
+			String identityJson = objectMapper.writeValueAsString(identityMap);
+			String data = "ENCRYPTED: " + identityJson;
+
+			IdaKycExchangeResponse idaKycExchangeResponse = new IdaKycExchangeResponse();
+			idaKycExchangeResponse.setEncryptedKyc(data);
+
+			IdaResponseWrapper<IdaKycExchangeResponse> idaResponseWrapper = new IdaResponseWrapper<>();
+			idaResponseWrapper.setResponse(idaKycExchangeResponse);
+			idaResponseWrapper.setTransactionID("TRAN123");
+			idaResponseWrapper.setVersion("VER1");
+
+			ResponseEntity<IdaResponseWrapper<IdaKycExchangeResponse>> responseEntity = new ResponseEntity<IdaResponseWrapper<IdaKycExchangeResponse>>(
+					idaResponseWrapper, HttpStatus.OK);
+
+			Mockito.when(tokenIdManager.generateTokenId(Mockito.anyString(), Mockito.anyString()))
+					.thenReturn("1122334455");
+			JWTSignatureResponseDto jwtSignatureResponseDto = Mockito.mock(JWTSignatureResponseDto.class);
+			Mockito.when(jwtSignatureResponseDto.getJwtSignedData()).thenReturn(data);
+			Mockito.when(signatureService.jwtSign(Mockito.any())).thenReturn(jwtSignatureResponseDto);
+			String encryptedIdentityData = "sample encrypted data";
+			Mockito.when(identityDataStore.getEncryptedIdentityData(Mockito.anyString(), Mockito.anyString()))
+					.thenReturn(encryptedIdentityData);
+
+			String decryptedIdentityData = CryptoUtil.encodeToURLSafeBase64(identityJson.getBytes());
+			Mockito.when(helperService.decrptData(encryptedIdentityData)).thenReturn(decryptedIdentityData);
+			Mockito.when(mapper.readValue(CryptoUtil.decodeURLSafeBase64(decryptedIdentityData), Map.class))
+					.thenReturn(identityMap);
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("name"))
+					.thenReturn(List.of("firstName", "lastName"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("address"))
+					.thenReturn(List.of("addressLine1", "city", "pinCode"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("firstName")).thenReturn(List.of("firstName"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("lastName")).thenReturn(List.of("lastName"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("addressLine1"))
+					.thenReturn(List.of("addressLine1"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("city")).thenReturn(List.of("city"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("pinCode")).thenReturn(List.of("pinCode"));
+			Mockito.when(idInfoHelper.getIdentityAttributesForIdName("dateOfBirth")).thenReturn(List.of("dateOfBirth"));
+
+			KycExchangeResult kycExchangeResult = idaAuthenticatorImpl.doKycExchange("relyingPartyId", "clientId",
+					kycExchangeDto);
+
+			Assert.assertEquals(idaKycExchangeResponse.getEncryptedKyc(), kycExchangeResult.getEncryptedKyc());
```

Comment:
@LoganathanSekar7627  as the logic to build the KYC is copy-pasted here, can we also assert the claims added in the KYC.

Reply:
Fixed.